reana_db/models.py

```diff
@@ -527,7 +533,8 @@ def __init__(
         self.git_repo = git_repo
         self.git_provider = git_provider
         self.restart = restart
-        self._run_number = self.assign_run_number(run_number)
+        self.generation_number = self.assign_generation_number(run_number)
+        self.restart_number = self.assign_restart_number(run_number)
```

Comment:
Given that these two functions are only called here, maybe we can merge them into a single one, so that we avoid calling `get_last_workflow` and querying the database twice? This new function would then return a tuple.